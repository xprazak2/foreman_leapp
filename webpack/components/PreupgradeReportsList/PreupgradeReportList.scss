@import '~@theforeman/vendor/scss/variables';

#preupgrade-report-entries-list-view {
  .empty-checkbox {
    content: '';
    width: 13px;
    height: 13px;
    content: " ";
    margin: 4px 0;
    display: block;
  }

  .sortable-header {
    &.active {
      color: $color-pf-blue-400;
    }

    &:hover {
      cursor: pointer;
    }

    .fa {
      margin-right: 10px;
    }
  }

  .list-view-pf-additional-info-item {
    flex: 1;
  }

  .list-view-header {
    border-top: 0px;
    background-color: #fff;
    border-left-color: #fff;
    border-right-color: #fff;

    .list-view-pf-description {
      font-size: 16px;
      font-weight: 600;
    }

    .list-view-pf-checkbox {
      margin-left: 17px;
    }
  }
}
