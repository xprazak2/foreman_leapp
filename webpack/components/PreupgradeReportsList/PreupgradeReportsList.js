import React from 'react';
import { ListView } from 'patternfly-react';

import PreupgradeReportEntry from './components/PreupgradeReportEntry'

const PreupgradeReportsList = props => {
  console.log(props);
  const flatEntries = props.preupgradeReports.reduce((memo, report) => [...memo, ...report.entries], [])

  return(
    <ListView>
      {flatEntries.map((entry, idx) => (
        <PreupgradeReportEntry
          entry={entry}
          key={idx}
        />
      ))}
    </ListView>
  );
}

export default PreupgradeReportsList;
