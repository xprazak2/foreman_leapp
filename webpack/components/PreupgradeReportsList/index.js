import React, { useState } from 'react';
import { ListView } from 'patternfly-react';
import PropTypes from 'prop-types';

import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
import { useForemanSettings } from 'foremanReact/Root/Context/ForemanContext';

import PreupgradeReportEntry from './components/PreupgradeReportEntry';
import PreupgradeReportsListHeader from './components/PreupgradeReportsListHeader';
import { entriesPage } from '../PreupgradeReports/PreupgradeReportsHelpers';

import './PreupgradeReportList.scss';

const PreupgradeReportsList = ({
  allEntries,
  isSelected,
  toggleSelected,
  sort,
  changeSort,
  toggleSelectAll,
}) => {
  const { perPage, perPageOptions } = useForemanSettings();
  const [pagination, setPagination] = useState({
    page: 1,
    perPage,
    perPageOptions,
  });

  return (
    <ListView id="preupgrade-report-entries-list-view">
      <PreupgradeReportsListHeader
        sort={sort}
        changeSort={changeSort}
        toggleSelectAll={toggleSelectAll}
      />
      {entriesPage(allEntries, pagination).map((entry, idx) => (
        <PreupgradeReportEntry
          entry={entry}
          key={idx}
          isEntrySelected={isSelected(entry)}
          toggleSelected={toggleSelected}
        />
      ))}
      <Pagination
        viewType="list"
        itemCount={allEntries.length}
        pagination={pagination}
        onChange={newPage => setPagination({ ...pagination, ...newPage })}
        dropdownButtonId="preupgrade-report-entries-pagination-dropdown"
      />
    </ListView>
  );
};

PreupgradeReportsList.propTypes = {
  allEntries: PropTypes.array.isRequired,
  isSelected: PropTypes.func.isRequired,
  toggleSelected: PropTypes.func.isRequired,
  sort: PropTypes.object.isRequired,
  changeSort: PropTypes.func.isRequired,
  toggleSelectAll: PropTypes.func.isRequired,
};

export default PreupgradeReportsList;
