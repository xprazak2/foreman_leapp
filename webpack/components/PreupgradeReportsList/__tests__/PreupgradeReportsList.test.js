import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import PreupgradeReportsList from '../index';

const allEntries = [
  { title: 'Fix me!', severity: 'Too severe to talk about' },
  { title: 'I am broken too', severity: 'medium' },
  { title: 'Octocat is not happy', severity: 'high' },
  { title: 'Not enough credits', severity: 'low' },
];

const isSelected = () => false;
const toggleSelected = () => {};
const sort = { attribute: '', order: 'asc' };
const changeSort = () => {};
const toggleSelectAll = () => {};

const fixtures = {
  'should render': {
    allEntries,
    fixAllWorking: false,
    isSelected,
    toggleSelected,
    sort,
    changeSort,
    toggleSelectAll,
  },
  'should render when working': {
    allEntries,
    fixAllWorking: true,
    isSelected,
    toggleSelected,
    sort,
    changeSort,
    toggleSelectAll,
  },
};

describe('PreupgradeReportsList', () =>
  testComponentSnapshotsWithFixtures(PreupgradeReportsList, fixtures));
