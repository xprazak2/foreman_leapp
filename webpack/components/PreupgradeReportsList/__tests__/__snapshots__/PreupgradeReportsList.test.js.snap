// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreupgradeReportsList should render 1`] = `
<ListView
  className=""
  id="preupgrade-report-entries-list-view"
>
  <PreupgradeReportsListHeader
    changeSort={[Function]}
    sort={
      Object {
        "attribute": "",
        "order": "asc",
      }
    }
    toggleSelectAll={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "Too severe to talk about",
        "title": "Fix me!",
      }
    }
    isEntrySelected={false}
    key="0"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "medium",
        "title": "I am broken too",
      }
    }
    isEntrySelected={false}
    key="1"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "high",
        "title": "Octocat is not happy",
      }
    }
    isEntrySelected={false}
    key="2"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "low",
        "title": "Not enough credits",
      }
    }
    isEntrySelected={false}
    key="3"
    toggleSelected={[Function]}
  />
  <PaginationWrapper
    dropdownButtonId="preupgrade-report-entries-pagination-dropdown"
    itemCount={4}
    onChange={[Function]}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
        "perPageOptions": Array [
          5,
          10,
          20,
          30,
          50,
        ],
      }
    }
    viewType="list"
  />
</ListView>
`;

exports[`PreupgradeReportsList should render when working 1`] = `
<ListView
  className=""
  id="preupgrade-report-entries-list-view"
>
  <PreupgradeReportsListHeader
    changeSort={[Function]}
    sort={
      Object {
        "attribute": "",
        "order": "asc",
      }
    }
    toggleSelectAll={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "Too severe to talk about",
        "title": "Fix me!",
      }
    }
    isEntrySelected={false}
    key="0"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "medium",
        "title": "I am broken too",
      }
    }
    isEntrySelected={false}
    key="1"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "high",
        "title": "Octocat is not happy",
      }
    }
    isEntrySelected={false}
    key="2"
    toggleSelected={[Function]}
  />
  <PreupgradeReportEntry
    entry={
      Object {
        "severity": "low",
        "title": "Not enough credits",
      }
    }
    isEntrySelected={false}
    key="3"
    toggleSelected={[Function]}
  />
  <PaginationWrapper
    dropdownButtonId="preupgrade-report-entries-pagination-dropdown"
    itemCount={4}
    onChange={[Function]}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
        "perPageOptions": Array [
          5,
          10,
          20,
          30,
          50,
        ],
      }
    }
    viewType="list"
  />
</ListView>
`;
